packages/plugins/quorum/src/deployer.ts

```diff
@@ -0,0 +1,92 @@
+import Web3 from 'web3';
+import { Embark, EmbarkEvents } from 'embark-core';
+import { Logger } from 'embark-logger';
+const embarkJsUtils = require('embarkjs').Utils;
+
+export default class QuorumDeployer {
+  private logger: Logger;
+  private events: EmbarkEvents;
+  private _web3: Web3 | null = null;
+  constructor(private embark: Embark) {
+    this.logger = embark.logger;
+    this.events = embark.events;
+  }
+
+  get web3() {
+    return (async () => {
+      if (!this._web3) {
+        const provider = await this.events.request2("blockchain:client:provider", "ethereum");
+        this._web3 = new Web3(provider);
+      }
+      return this._web3;
+    })();
+  }
+
+  public registerDeployer() {
+    this.events.request("deployment:deployer:register", "ethereum", this.deployer.bind(this));
+    this.embark.registerActionForEvent("deployment:contract:shouldDeploy", { priority: 40 }, this.skipBytecodeCheck.bind(this));
+  }
+
+  skipBytecodeCheck(params, cb) {
+    const skipBytecodeCheck = this.embark.config.contractsConfig?.contracts[params.contract.className]?.skipBytecodeCheck ?? false;
+    if (skipBytecodeCheck) {
+      this.logger.warn(__("WARNING: Skipping bytecode check for %s deployment. Performing an embark reset may cause the contract to be re-deployed to the current Quorum node regardless if it was already deployed on another node.", params.contract.className));
+    }
+    params.skipBytecodeCheck = skipBytecodeCheck;
+    cb(null, params);
+  }
+
+  async deployer(contract, done) {
+    try {
+      const web3 = await this.web3;
+      const [account] = await web3.eth.getAccounts();
+      const contractObj = new web3.eth.Contract(contract.abiDefinition, contract.address);
+      // @ts-ignore
+      contractObj._jsonInterface.find(obj => {
+        if (obj.type === 'constructor') {
+          if (!obj.name) {
+            // Add constructor as the name of the method, because if there is an error, it prints `undefined` otherwise
+            obj.name = 'constructor';
+          }
+          return true;
+        }
+        return false;
+      });
+      const code = contract.code.substring(0, 2) === '0x' ? contract.code : "0x" + contract.code;
+      const contractObject = contractObj.deploy({ arguments: (contract.args || []), data: code });
+      if (contract.gas === 'auto' || !contract.gas) {
+        const gasValue = await contractObject.estimateGas({ value: 0, from: account });
+        const increase_per = 1 + (Math.random() / 10.0);
+        contract.gas = Math.floor(gasValue * increase_per);
+      }
+
+      if (!contract.gasPrice) {
+        contract.gasPrice = await web3.eth.getGasPrice();
+      }
+
+      embarkJsUtils.secureSend(web3, contractObject, {
+        from: account,
+        gas: contract.gas,
+        gasPrice: contract.gasPrice,
+        privateFor: contract.privateFor,
+        privateFrom: contract.privateFrom,
+        bytecodeWithInitParam: contractObject.encodeABI()
```

Comment:
Correct, we are simply adding three parameters to the deployment. I originally had it this way (adding the three parameters to the one single deployer) but in the end felt that adding superfluous parameters that may not be needed is not a very clean approach. 

Instead of leaving `privateFor/privateFrom` in the original deployer, I've opted to update this so that we now update `params.additionalDeployParams` in the the action `deployment:contract:beforeDeploy`. These additional deployment parameters are passed to the core deployer and used in the deployment. This isolates all additional needed deployment parameters to the quorum plugin without polluting core embark functionality.